Kerberos.NET/Replay/MemoryTicketCache.cs

```diff
@@ -51,11 +54,14 @@ protected override async Task BackgroundCacheOperation()
             }
         }
 
-        public override Task<bool> AddAsync(TicketCacheEntry entry)
+        public override ValueTask<bool> AddAsync(TicketCacheEntry entry)
         {
-            var added = this.Add(entry);
+            if (this.Add(entry))
+            {
+                return TrueTask;
+            }
 
-            return Task.FromResult(added);
+            return FalseTask;
```

Comment:
With the ValueTasks this caching isn't needed -- sorry for being unclear with the comment about this.
For `Task` it has the benefit of not allocating. A ValueTask is a struct, so this problem doesn't occur.